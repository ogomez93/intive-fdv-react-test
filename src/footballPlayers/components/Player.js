import React from 'react';
import moment from 'moment';

export function calculateAge(dob) {
  let date = moment(dob, "YYYY-MM-DD").toDate();
  let today = new Date();
  let timeDiff = Math.abs(today.getTime() - date.getTime());
  let age = Math.ceil(timeDiff / (1000 * 3600 * 24 * 365)) - 1;
  return age;
}

const Player = ({ player }) => (
  <tr>
    <td label="Name" className="border-2 border-black px-1 py-3 sm:py-6">{player.name}</td>
    <td label="Position" className="border-2 border-black px-1 py-3 sm:py-6">{player.position}</td>
    <td label="Nationality" className="border-2 border-black px-1 py-3 sm:py-6">{player.nationality}</td>
    <td label="Age" className="border-2 border-black px-1 py-3 sm:py-6">{calculateAge(player.dateOfBirth)}</td>
  </tr>
);

export default Player;