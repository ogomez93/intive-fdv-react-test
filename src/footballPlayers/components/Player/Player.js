import React from 'react';
import calculateAge from 'footballPlayers/utils/calculateAge';

const Player = ({ dateOfBirth, name, nationality, position }) => (
  <tr>
    <td label="Name" className="border-2 border-black px-1 py-3 sm:py-6">
      {name}
    </td>
    <td label="Position" className="border-2 border-black px-1 py-3 sm:py-6">
      {position}
    </td>
    <td label="Nationality" className="border-2 border-black px-1 py-3 sm:py-6">
      {nationality}
    </td>
    <td label="Age" className="border-2 border-black px-1 py-3 sm:py-6">
      {calculateAge(dateOfBirth)}
    </td>
  </tr>
);

export default Player;
