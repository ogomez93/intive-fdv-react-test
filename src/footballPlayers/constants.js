export const NAME = 'footballPlayers';
export const PLAYERS_API = 'https://football-players-b31f2.firebaseio.com/players.json?print=pretty';