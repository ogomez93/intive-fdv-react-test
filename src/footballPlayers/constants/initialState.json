{
  "fetching": true,
  "players": [],
  "nameFilter": "",
  "ageFilter": "",
  "positionFilter": ""
}
