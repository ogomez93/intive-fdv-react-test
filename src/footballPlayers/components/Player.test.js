import React from 'react';
import renderer from 'react-test-renderer';

import Player, { calculateAge } from './Player';
import { presentYear, eightteenYearsAgo, player } from './test/constants';

describe('Component: <Player />', () => {
  const component = renderer.create( <Player player={player} /> );
  const playerProps = component.toJSON().children;
  
  test('renders the component', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('component renders player name correctly', () => {
    const nameTd = playerProps.find(p => p.props.name === 'name');
    expect(nameTd.children[0]).toEqual('test name');
  });

  test('component renders player position correctly', () => {
    const positionTd = playerProps.find(p => p.props.name === 'position');
    expect(positionTd.children[0]).toEqual('test position');
  });

  test('component renders player nationality correctly', () => {
    const nationalityTd = playerProps.find(p => p.props.name === 'nationality');
    expect(nationalityTd.children[0]).toEqual('test nationality');
  });

  test('component renders player age correctly', () => {
    const ageTd = playerProps.find(p => p.props.name === 'age');
    expect(ageTd.children[0]).toEqual('18');
  });
});

describe('Function: calculateAge(dateOfBirth)', () => {
  test('someone born this year is 0 years old', () => {
    expect(calculateAge(`${presentYear}-01-01`)).toEqual(0);
  });

  test('someone born 18 years ago is 18 years old', () => {
    expect(calculateAge(`${eightteenYearsAgo}-01-01`)).toEqual(18);
  });
});
