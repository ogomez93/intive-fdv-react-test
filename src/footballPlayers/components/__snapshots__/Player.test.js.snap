// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Player /> renders the component 1`] = `
<tr>
  <td
    className="border-2 border-black px-1 py-3 sm:py-6"
    label="Name"
  >
    test name
  </td>
  <td
    className="border-2 border-black px-1 py-3 sm:py-6"
    label="Position"
  >
    test position
  </td>
  <td
    className="border-2 border-black px-1 py-3 sm:py-6"
    label="Nationality"
  >
    test nationality
  </td>
  <td
    className="border-2 border-black px-1 py-3 sm:py-6"
    label="Age"
  >
    18
  </td>
</tr>
`;
